dataactbroker/handlers/fileHandler.py

```diff
@@ -552,6 +552,9 @@ def load_d_file(self, url, upload_name, timestamped_name, job_id, is_local, resp
 
             logger.debug('Marking job id of %s', job_id)
             mark_job_status(job_id, "finished")
+            # only need to close it if we threaded it in the first place
+            if not self.isLocal:
```

Comment:
You could put this in a finally block for the try/except and have a single point of closure instead of duplicating the check in 2 places. Also, I don't think we need to close only if it's non-local. If we close regardless of whether or not we threaded, any subsequent access to GlobalDB would just grab another session.